Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM mcr.microsoft.com/dotnet/sdk:8.0 AS build
+WORKDIR /app
+
+# copy csproj and restore as distinct layers
+COPY *.sln .
+COPY ./src/Planthor.IdentityServerAspNetIdentity/*.csproj ./src/Planthor.IdentityServerAspNetIdentity/
```

Comment:
Maybe we need something better than the hard-coded (a loop based on *.csproj pattern?)

Reply:
In Docker, there’s no built-in way to use a wildcard pattern to copy multiple directories in a loop I think.
But we can use a workaround by copying all the .csproj files to a temporary directory... 
This approach is more flexible, I am not sure could take all .csproj files best practices...  But I will put my solution here:
```Dockerfile
# Copy csproj files to a temporary directory and restore as distinct layers.
COPY *.sln .
COPY */*.csproj ./temp/
RUN for file in $(ls ./temp/*.csproj); do mkdir -p ./${file%.*}/ && mv $file ./${file%.*}/; done
RUN dotnet restore

```